app/routes/image-set.js

```diff
@@ -15,5 +15,45 @@ export default Ember.Route.extend( OrganizationRouteMixin, RequireUserMixin, {
 
   model: function(params) {
     return this.store.find('imageSet', params.image_set_id);
+  },
+
+  actions: {
+    associate: function(imageSet, lion) {
+      var route = this,
+          previousIsVerified = imageSet.get('isVerified');
+
+      imageSet.setProperties({
+        lion: lion,
+        isVerified: false
+      });
+
+      imageSet.save().then(function() {
+        route.transitionTo('lion', lion);
+      }, function(){
+        imageSet.setProperties({
+          lion: null,
+          isVerified: previousIsVerified
+        });
+      });
+    },
+
+    createLion: function(imageSet, lionName){
+      var route = this;
+
+      var lion = this.store.createRecord('lion', {
+        primaryImageSet: imageSet,
+        name: lionName
+      });
+
+      // Bind lion to controller so that we can show errors
+      var controller = this.get('controller');
+      controller.set('newLion', lion);
+
+      lion.save().then(function(lion){
+        route.transitionTo('lion', lion);
+      }, function(){
+        //error shown in template
```

Comment:
technically you probably want to ensure that the error is `instanceof DS.InvalidError` -- the error could be something more generic like a syntax error or network failure, in which case you'd want to `return` the error and let somethign else handle it